assets/app/scripts/services/logs/logLinks.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .factory('logLinks', [
+    '$anchorScroll',
+    '$document',
+    '$location',
+    '$window',
+    function($anchorScroll, $document, $location, $window) {
+      // TODO (bpeterse): a lot of these functions are generic and could be moved/renamed to
+      // a navigation oriented service.
+      var doc = $document[0];
+
+      var createObjectURL = function() {
+        return (window.URL || window.webkitURL || {}).createObjectURL || _.noop;
+      };
+
+      var revokeObjectURL = function() {
+        return (window.URL || window.webkitURL || {}).revokeObjectURL || _.noop;
+      };
+
+      var canDownload = function() {
+        return !!createObjectURL();
+      };
+
+      var makeDownload = function(obj, filename) {
+        obj = _.isString(obj) ? obj : JSON.stringify(obj);
+        var a = doc.createElement('a');
+        a.href = createObjectURL()(new Blob([obj], {type: 'text/plain'}));
+        a.download = (filename || 'download') + '.txt';
+        doc.body.appendChild(a);
+        a.click();
+        revokeObjectURL()(a.href);
+        doc.body.removeChild(a);
+      };
+
+      var scrollTop = function() {
+        $window.scrollTo(null, 0);
+      };
+
+      var scrollBottom = function() {
+        $window.scrollTo(null, $document.innerHeight());
+      };
+
+      var scrollTo = function(anchor, event) {
+        // sad face. stop reloading the page!!!!
+        event.preventDefault();
+        event.stopPropagation();
+        $location.hash(anchor);
+        $anchorScroll(anchor);
+      };
+
+      var fullPageLink = function() {
+         $location
+          .path($location.path())
+          .search(
+            angular.extend($location.search(), {
+              view: 'chromeless'
+          }));
```

Comment:
I think this can simply be

```
$location.search('view', 'chromeless');
```

If I'm reading the doc right, it doesn't overwrite other params.

https://docs.angularjs.org/api/ng/service/$location